src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1389,6 +1391,7 @@ private function addHttpClientSection(ArrayNodeDefinition $rootNode)
                             ->end()
                         ->end()
                         ->arrayNode('scoped_clients')
+                            ->performNoDeepMerging()
```

Comment:
why not allow deep merge for clients, eg if one wants one more option for the test env?

Reply:
but what if a client wants less options?

Reply:
it should be still possible, by properly using dev/test/prod + all-envs files,no?

Reply:
if deep merging happens and one needs less options for a given client, then it requires to reset the options to their default values and, if one of those options is scalar and does not have a default value (e.g. `max_redirects`) then you'll get an invalid type exception if you try resetting it with `null`. We would better force copy-pasting I think.

Reply:
I'm sorry but I don't think this is what we want:
the reported issue happens when array prototypes are found inside array prototypes.
When this happens, we don't want the nested "senders" prototype to be deep merged, I agree with that.
So to me, the `performNoDeepMerging` should be added to the nested prototype, not to its parent.
Same for other options, case by case.
E.g. for http_client, the "query" option should not be deep merged. But the "resolve" one could be, etc.